pwiz_tools/Skyline/Controls/Graphs/GraphFullScan.cs

```diff
@@ -598,13 +604,125 @@ private void PopulateProperties()
                     var chromSet = stateProvider.DocumentUI.Settings.MeasuredResults.Chromatograms.FirstOrDefault(
                         chrom => chrom.ContainsFile(_msDataFileScanHelper.ScanProvider.DataFilePath));
                     spectrumProperties.ReplicateName = chromSet?.Name;
+                    if (_peaks != null && _peaks.Length > 0)
+                    {
+                        var nodePath = DocNodePath.GetNodePath(_msDataFileScanHelper.CurrentTransition?.Id,
+                            _documentContainer.DocumentUI);
+
+                        if (!nodePath.Precursor.Id.Equals(_precursor?.DocNode?.Id))
+                            _precursor = new GraphSpectrum.Precursor(_documentContainer.DocumentUI.Settings, null,

```

Comment:
(Not important)
This should have curly braces around it. Whenever the block of an "if" statement is more than one line of code, either because it's a long statement or because there's a comment, it should be surrounded by curly braces.